File: oppgave-03/src/test/java/no/spk/workshop/kotlin/LegacyGreetingProviderTest.java
package no.spk.workshop.kotlin;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class LegacyGreetingProviderTest {

    @Test
    @DisplayName("Provides correct greeting")
    void providesCorrectGreeting() {
        final GreetingProvider greetingProvider = new LegacyGreetingProvider("Java");
        final String greeting = greetingProvider.greeting();
        assertThat(greeting).isEqualTo("Hello, Java!");
    }
}

File: oppgave-02/src/test/kotlin/no/spk/workshop/kotlin/LegacyGreetingProviderTest.kt
package no.spk.workshop.kotlin

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

internal class KotlinGreetingProviderTest {

    @Test
    fun `Provides correct greeting`() {
        val greetingProvider = KotlinGreetingProvider("Kotlin")
        val greeting = greetingProvider.greeting()
        greeting shouldBe "Hello, Kotlin!"
    }
}
